### PR TITLE
add payment processing to case attachment

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-864

### Change description ###

add payment processing to case attachment

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 1]
```diff
@@ -264,16 +280,26 @@ private void attachCaseByCcdId(
...
-        log.info("Attached exception record '{}' to case with CCD ID '{}'", exceptionRecordId, targetCaseCcdRef);
+        log.info(
+            "Successfully attach Exception Record with ID {} to case with id {}",
+            theCase.getId(),
+            exceptionRecord.getId()
+        );
+
+        paymentsProcessor.updatePayments(exceptionRecord, theCase.getId());
+
+        log.info("Exception record attached, exception id {}, case id {}",   theCase.getId(), exceptionRecord.getId());
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: these 2 log entries look very similar in purpose

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java [hunk 2]
```diff
@@ -148,6 +148,11 @@ public void should_attach_exception_record_to_case_by_attach_to_case_reference()
         verifyExceptionRecordAttachesToCase(null);
     }
 
+    @Test
+    public void should_attach_exception_record_to_case_by_ccd_search_case_reference_and_payment() throws Exception {
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java]: may be should attach exception record to case by search case ref and publish payments ?

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java [hunk 3]
```diff
@@ -176,6 +181,25 @@ private void verifyExceptionRecordAttachesToCase(String searchCaseReferenceType)
...
+        // when
+        invokeCallbackEndpoint(caseDetails, exceptionRecord, searchCaseReferenceType, true);
+
+        //then
+        await("Exception record is attached to the case")
+            .atMost(60, TimeUnit.SECONDS)
+            .pollDelay(2, TimeUnit.SECONDS)
+            .until(() -> isExceptionRecordAttachedToTheCase(caseDetails, 1));
+
+        verifyExistingCaseIsUpdatedWithExceptionRecordData(caseDetails, exceptionRecord, 1);
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java]: not sure why this needs to be extracted can be in the same test method ?

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java [hunk 4]
```diff
@@ -242,6 +277,12 @@ private void invokeCallbackEndpoint(
             exceptionRecordData.put("searchCaseReference", searchCaseReference);
         }
 
+        if (containPayments) {
+            exceptionRecordData.put("containsPayments", "Yes");
+            exceptionRecordData.put("envelopeId", "21321931312-32121-312112");
+            exceptionRecordData.put("poBoxJurisdiction", "sample jurisdiction");
+        }
+
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java]: not sure why we need this to be added to exception record ?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsProcessor.java [hunk 5]
```diff
@@ -60,11 +60,12 @@ public void updatePayments(CaseDetails exceptionRecord, long newCaseId) {
             );
 
         if (containsPayments) {
+
+            log.info("Contains Payments, send payment update message. ER id: {}", exceptionRecord.getId());
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsProcessor.java]: Also can we please add jurisdiction payment Dcn to this log ?

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java (hunk 4 above)
Comment (mdayican) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java]: we publish envelopeId and jurisdiction.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsProcessor.java (hunk 5 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsProcessor.java]: do we always have payment Dcns ?

**Pull request merged**